lib/collection/cookie.js

```diff
@@ -157,15 +157,15 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
              * host name that is sending the Set-Cookie header. The SDK does not perform this check, but the underlying
              * client that actually sends the request could do it automatically.
              */
-            domain: options.domain,
+            domain: _.isString(options.domain) ? options.domain : undefined,
 
             /**
              * @type {String}
              *
              * @note
              * On server, the default value for the path option is the path of the URL that sent the Set-Cookie header.
              */
-            path: options.path,
+            path: _.isString(options.path) ? options.path : undefined,
```

Comment:
Let's add tests to check behaviour when `path` and `domain` are not `strings`. Ideally we should be checking if they are:
1. `null`,
2. `object`,
3. `array`,
4. `number`,
5. `function`
6. empty `string` (in this case it should be set to empty string)

Reply:
Fixed, added for both `domain` and `path`